homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -108,6 +115,58 @@ def get_platforms(config_entry):
     return []
 
 
+def _async_update_data_default(hass, device):
+    async def update():
+        """Fetch data from the device using async_add_executor_job."""
+        try:
+            async with async_timeout.timeout(10):
+                state = await hass.async_add_executor_job(device.status)
+                _LOGGER.debug("Got new state: %s", state)
+                return state
+
+        except DeviceException as ex:
+            raise UpdateFailed(ex) from ex
+
+    return update
+
+
+def _async_update_data_vacuum(hass, device: Vacuum):
+    def update():
+        data = {
+            KEY_VACUUM_STATUS: device.status(),
+            KEY_VACUUM_DND_STATUS: device.dnd_status(),
+            KEY_VACUUM_LAST_CLEAN_STATUS: device.last_clean_details(),
+            KEY_VACUUM_CONSUMABLE_STATUS: device.consumable_status(),
+            KEY_VACUUM_CLEAN_HISTORY_STATUS: device.clean_history(),
+        }
+
```

Comment:
I don't think it's "homeassistant-esque", but if it's allowed by the coding standards / reviewers, I would rather get rid of the overuse of consts inside the integration, and rather use a dataclass-based container for the coordinator data.

This would allow easier extending later on thanks to type hints:
```
@dataclass
class VacuumCoordinatorData:
    status: VacuumStatus
    dnd_status: VacuumDnDStatus
    ...
```